Jenkinsfile

```diff
@@ -396,6 +418,11 @@ pipeline {
                                     test-integration
                                 '''
                             }
+                            post {
+                                always {
+                                    junit testResults: 'bundles/**/*-report.xml', allowEmptyResults: true
```

Comment:
I believe this glob pattern is what's causing the duplicate results. There is a `latest` symlink that ends up going down a rabbit hole.

See one of the [artifacts](https://ci.docker.com/public/blue/organizations/jenkins/moby/detail/PR-39789/4/artifacts) that are saved with the same glob pattern:
```
bundles/latest/latest/latest/latest/latest/test-integration/amd64-integration-build-junit-report.xml
```

I suggest changing the glob pattern to `bundles/test-integration/*-junit-report.xml` like I did for this test job that just collects tests for `integration`: https://ci.docker.com/public/job/moby/job/PR-39789/6/testReport/

Reply:
LOL, wow, nice find! I didn't expect it to be recursing like that (although I've had cases where my IDE wasn't fond of it and got into a loop).

Using `bundles/test-integration/*-junit-report.xml` won't work for the Janky tests, where `1/`, `2/`, `3/` subdirectories are used currently. We could probably work around that, but I was trying to make the steps as generic as possible.

With the original reason for that symlink (https://github.com/moby/moby/pull/12049) no longer being in place (https://github.com/moby/moby/pull/34685), I think it should be safe to remove it, as it no longer serves any purpose.

Pushing a commit to remove it (and to see if that removes the problem), but will open it as a separate PR for visibility

Reply:
Opened https://github.com/moby/moby/pull/39795 with just that commit